packages/currency/src/currency-utils.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Checks if a Near address is valid according to a currency network.
+ * Returns true if the currency network is not given and the address is correct for any network.
+ */
+export const isValidNearAddress = (address: string, network?: string): boolean => {
+  if (!network) {
+    return isValidNearAddress(address, 'aurora') || isValidNearAddress(address, 'aurora-testnet');
+  }
+  if (!address.match(/^(([a-z\d]+[-_])*[a-z\d]+\.)*([a-z\d]+[-_])*[a-z\d]+$/)) {
```

Comment:
this deserves a comment :)

Reply:
This regex was moved from the `near-native.ts` implementation: https://github.com/RequestNetwork/requestNetwork/blob/d26ef8beea87e9ca268689495143222c6b8d034b/packages/advanced-logic/src/extensions/payment-network/near-native.ts#L40
; but I don't know where it comes from exactly. I didn't find much content on it on the web.
Maybe @yomarion can help on this?

Reply:
See with @yomarion, added comments with NEAR specification link